### ISSUE TITLE
False positive `E1101` when inheriting from tuple and using sorted function inside __new__ dunder

### ISSUE THREAD

**Issue by Migoyan** [pylint-dev/pylint#8876]
### Bug description

When inheriting from tuple and using `sorted` function, any added attributes aren't recognized anymore by pylint
```py
# pylint: disable=missing-module-docstring
# pylint: disable=missing-class-docstring
# pylint: disable=missing-function-docstring


class Foo(tuple):

    def __new__(cls, iterable) -> 'Foo':
        instance = super().__new__(cls, iterable)

        instance._x = 0
        instance.y = 1
        return instance

    @property
    def x(self):
        return self._x


my_foo = Foo(("c", "b", "a"))
print(my_foo.x, my_foo.y)


class Bar(tuple):

    def __new__(cls, iterable) -> 'Bar':
        instance = super().__new__(cls, sorted(iterable))

        instance._x = 0
        instance.y = 1
        return instance

    @property
    def x(self):
        return self._x


my_bar = Bar(("c", "b", "a"))
print(my_bar.x, my_bar.y)

```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:35:15: E1101: Instance of 'Bar' has no '_x' member; maybe 'x'? (no-member)
test.py:39:16: E1101: Instance of 'Bar' has no 'y' member; maybe 'x'? (no-member)

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 5.00/10, +0.00)
```

### Expected behavior

No errors should be reported

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.6
Python 3.7.3 (default, Jun 29 2023, 18:03:57)
```

### OS / Environment

Debian 10

**Comment by nickdrozd** [comment]
I don't really understand how the `Foo` examples works. But since Pylint accepts it, it seems like it ought to accept the `Bar` example too.

One difference is that `sorted` does not return a `tuple` back:

```
>>> type(("c", "b", "a"))
<class 'tuple'>
>>> type(sorted(("c", "b", "a")))
<class 'list'>
```

Maybe that has something to do with it?

**Comment by Migoyan** [comment]
> One difference is that `sorted` does not return a `tuple` back:
> 
> ```
> >>> type(("c", "b", "a"))
> <class 'tuple'>
> >>> type(sorted(("c", "b", "a")))
> <class 'list'>
> ```
> 
> Maybe that has something to do with it?

Thank you for your answer. I've done more tests (code below) considering what you've told me.
```py
# pylint: disable=missing-module-docstring
# pylint: disable=missing-class-docstring
# pylint: disable=missing-function-docstring


class Foo(list):

    def __new__(cls, iterable) -> 'Foo':
        instance = list.__new__(cls, iterable)

        instance._x = 0
        instance.y = 1
        return instance

    @property
    def x(self):
        return self._x


my_foo = Foo(("c", "b", "a"))
print(my_foo.x, my_foo.y)


class Bar(tuple):

    def __new__(cls, iterable) -> 'Bar':
        instance = super().__new__(cls, sorted(iterable))

        instance._x = 0
        instance.y = 1
        return instance

    @property
    def x(self):
        return self._x


my_bar = Bar(("c", "b", "a"))
print(my_bar.x, my_bar.y)


class Bar1(list):

    def __new__(cls, iterable) -> 'Bar1':
        instance = list.__new__(cls, sorted(iterable))

        instance._x = 0
        instance.y = 1
        return instance

    @property
    def x(self):
        return self._x


my_bar_1 = Bar1(("c", "b", "a"))
print(my_bar_1.x, my_bar_1.y)


class Bar2(tuple):

    def __new__(cls, iterable) -> 'Bar2':
        instance = super().__new__(cls, tuple(sorted(iterable)))

        instance._x = 0
        instance.y = 1
        return instance

    @property
    def x(self):
        return self._x


my_bar_2 = Bar2(("c", "b", "a"))
print(my_bar_2.x, my_bar_2.y)


class Bar3(tuple):

    def __new__(cls, iterable) -> 'Bar3':
        instance = super().__new__(cls, list(sorted(iterable)))

        instance._x = 0
        instance.y = 1
        return instance

    @property
    def x(self):
        return self._x


my_bar_3 = Bar3(("c", "b", "a"))
print(my_bar_3.x, my_bar_3.y)
```

Here the results of pylint
```
************* Module test
test.py:35:15: E1101: Instance of 'Bar' has no '_x' member; maybe 'x'? (no-member)
test.py:39:16: E1101: Instance of 'Bar' has no 'y' member; maybe 'x'? (no-member)
test.py:53:15: E1101: Instance of 'Bar1' has no '_x' member; maybe 'x'? (no-member)
test.py:57:18: E1101: Instance of 'Bar1' has no 'y' member; maybe 'x'? (no-member)

------------------------------------------------------------------
Your code has been rated at 6.00/10 (previous run: 5.00/10, +1.00)
```
Inheriting from `list` instead of `tuple` doesn't solve the issue. On contrary using `tuple` or `list` built-in does solve it. I'm not aware how pylint works under the hood to understand why it's different.
